## Assignment 3

This assignmemt is based on content discussed in module 3 and test basic concepts of probability theory and normalization in statistics.

## Learning outcomes

-   Work on problems of different distributions eg., binomial, gaussian 
-   Calculate z score 
-	 Make statistical inferences on given data
-	 Construct a null and an alternate hypothesis
-	 Find the p-value for a given hypothesis and T test statistic.


**Question 1**

The Capital Asset Pricing Model (CAPM) is a financial model that assumes returns on a portfolio are normally distributed.  Suppose a portfolio has an average annual return of 14.7% (i.e., an average gain on 14.7%) with a standard deviation of 33%.  A return of 0% means the value of the portfolio doesn't change, a negative return means that the portfolio loses money, and a positive return means that the portfolio gains money. Determine the following:

1. What percentage of years does this portfolio lose money, (i.e. have a return less than 0%)?
2. What is the cutoff for the highest 15% of annual returns with this portfolio?

See CAPM here https://en.wikipedia.org/wiki/Capital_asset_pricing_model 

In [2]:
import scipy.stats as st
import math

In [3]:
st.norm(14.7,33).cdf(0)

0.3279956507031998

32.8% of years will see the portfolio lose money

In [5]:
st.norm(14.7, 33).ppf(0.85)

48.90230185329506

15% of returns will be greater than or equal to 48.9%

**Question 2**

Past experience indicates that because of low morale, a company loses 20 hours a year per employee due to lateness and abstenteeism.  Assume that the standard deviation of the population is 6 and normally distributed.

The HR department implemented a new rewards system to increase employee morale, and after a few months it collected a random sample of 20 employees and the annualized absenteeism was 14.

1. Could you confirm that the new rewards system was effective with a 90% confidence?
2. An HR subject matter expert would be very happy if the program could reduce absenteeism by 20% (i.e. to 16 hours).  Given the current sampling parameters, what is the probability that the new rewards system reduced absenteeism to 16 hours and you miss it?
3. Repeat part 1) and 2) with an α = 95% CI.
4. Based on the answers in 3), is the sampling method good enough to identify a reduction from 20 to 16 hours if I use a confidence of 95%?
5. What should the sample size be if you want β to be 5%

In [11]:
#1. Could you confirm that the new rewards system was effective with a 90% confidence?
#comparing 14 to old value, 20
#std error = 6/sqrt20
#DF = 20-1 = 19
(st.t.cdf(((14-20)/(6/math.sqrt(20))),df=19))*100

0.013059668917013078

Significance level is 1 - Confidence = 0.10. We cannot confirm the new rewards system was effective with a 90% confidence, p value is too low compared to significance

In [39]:
#2. An HR subject matter expert would be very happy if the program could reduce absenteeism by 20% (i.e. to 16 hours).  Given the current sampling parameters, what is the probability that the new rewards system reduced absenteeism to 16 hours and you miss it?
1 - st.t.pdf(((16-20)/(6/math.sqrt(20))),df=19)

0.9915194858448683

It is extremely unlikely that we missed the 16 hour absenteeism with the current sampling parameters. Our p value is greater than the confidence which means we accept the hypothesis that the new system is not effective. 

#3. Repeat part 1) and 2) with an α = 95% CI.

#4. Based on the answers in 3), is the sampling method good enough to identify a reduction from 20 to 16 hours if I use a confidence of 95%?

The probabilities are the same as above, however these values are still not good enough to work with a 95% CI either as p values are still not able to beat significance  

#5. What should the sample size be if you want β to be 5%

In [40]:
p = st.t.cdf(((14-20)/(6/math.sqrt(20))),df=19)
z = st.t.ppf(p, df = 19)
z_beta = st. t.cdf((1-0.05), df = 19)

In [42]:
sample_size = ((z_beta + z)**2)*(6**2)/20**2
sample_size

1.1984708372323225

For a sample that works with beta being less than 5%, the sample size should be around 1.

**Question 3**

Chi-Square Goodness of fit

Please access and review **section 6.3.5** in the OpenIntro Statistics textbook:

Diez, D., Barr, C. & Çetinkaya-Rundel, M. (2017). OpenIntro Statistics (3rd Ed.). https://www.openintro.org/stat/textbook.php?stat_book=os

Given the information in section 6.3.5, write python code for the following:

 - Calculate the expected values based on the geometric distribution with a probability of 53.2%
 - Compare the expected vs. the observed values from the textbook using the Chi-Square distribution
 - Reach a conclusion
 - Explain what is the business impact of your conclusion

In [44]:
#your work here
import pandas as pd
obs = {'DAYS':[1,2,3,4,5,6,7], "OBSERVED":[717,369,155,69,28,14,10]}
df_observe = pd.DataFrame(obs)
df_observe

DAYS  OBSERVED
0     1       717
1     2       369
2     3       155
3     4        69
4     5        28
5     6        14
6     7        10

In [52]:
for i in range(1,8):
    print("Expected value for day ",i," is:", (((1-0.532)**(i-1))*(0.532))*1362)

Expected value for day  1  is: 724.5840000000001
Expected value for day  2  is: 339.105312
Expected value for day  3  is: 158.70128601599998
Expected value for day  4  is: 74.27220185548799
Expected value for day  5  is: 34.75939046836838
Expected value for day  6  is: 16.2673947391964
Expected value for day  7  is: 7.613140737943915


In [54]:
exp = {'DAYS':[1,2,3,4,5,6,7], "OBSERVED":[725,339,159,74,35,16,8]}
df_expect = pd.DataFrame(exp)
df_expect

DAYS  OBSERVED
0     1       725
1     2       339
2     3       159
3     4        74
4     5        35
5     6        16
6     7         8

In [58]:
st.chi2_contingency(df_observe)

(261.63674099269497,
 1.3342906314572928e-53,
 6,
 array([[1.44633094e+01, 7.03536691e+02],
        [7.47338129e+00, 3.63526619e+02],
        [3.18273381e+00, 1.54817266e+02],
        [1.47050360e+00, 7.15294964e+01],
        [6.64748201e-01, 3.23352518e+01],
        [4.02877698e-01, 1.95971223e+01],
        [3.42446043e-01, 1.66575540e+01]]))

In [57]:
st.chi2_contingency(df_expect)

(264.1192095146861,
 3.9294282492745334e-54,
 6,
 array([[1.46878613e+01, 7.11312139e+02],
        [6.89884393e+00, 3.34101156e+02],
        [3.27745665e+00, 1.58722543e+02],
        [1.57803468e+00, 7.64219653e+01],
        [8.09248555e-01, 3.91907514e+01],
        [4.45086705e-01, 2.15549133e+01],
        [3.03468208e-01, 1.46965318e+01]]))

Both the expected and observed p values are far less than 0.05, so we can reject the null hypothesis. The business impact of this is that stock activity each day is not independent of the stock's behaviour the previous day, which means there are trends that can be analyzed in trading. 